connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -777,6 +783,30 @@ private void startTask(ConnectorTaskId taskId) {
         worker.startTask(taskId, taskConfig, connConfig, this, initialState);
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State action) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (action) {
+                    case UNASSIGNED:
```

Comment:
Using `UNASSIGNED` might be confusing here since the connector/tasks have been assigned at this point. I think we should either stick to current state or target state here even though it isn't user visible as it'll make it clearer what's going on. (Relatedly, but not really important, `action` might be better named `targetState` or something like that given it is using the `TaskStatus.State` type.)